src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.EventDispatch.cs

```diff
@@ -77,21 +79,26 @@ private Task ReturnErrorResponse(string detailedMessage)
             : Task.CompletedTask;
     }
 
-    private async Task SetNotFoundResponseAsync(string baseUri)
+    internal async Task SetNotFoundResponseAsync(string baseUri, NotFoundEventArgs args)
     {
-        if (_httpContext.Response.HasStarted)
+        if (_httpContext.Response.HasStarted ||
+            // POST waits for quiescence -> rendering the NotFoundPage would be queued for the next batch
+            // but we want to send the signal to the renderer to stop rendering future batches -> use client rendering
+            string.Equals(_httpContext.Request.Method, "POST", StringComparison.OrdinalIgnoreCase))
```

Comment:
HttpMethods.IsPost(_httpContext.Request.Method)

Reply:
ToDo: `EndpointHtmlRenderer.Streaming` in `WriteResponseTemplate` also uses the same way, refector both.